cmd/bundle/main.go

```diff
@@ -154,3 +178,163 @@ func resolveAuthBasedOnTarget(ref name.Reference) (authn.Authenticator, error) {
 		RegistryToken: authConfig.RegistryToken,
 	}), nil
 }
+
+// Prune removes the image from the container registry
+//
+// Deleting a tag, or a whole repo is not as straightforward as initially
+// planned as DockerHub seems to restrict deleting a single tag for
+// standard users. This might be subject to change, but as of September
+// 2021 it is limited to the business tier. However, there is an API call
+// to delete the whole repository. In case there is only one tag used in
+// a repository, the effect is pretty much the same. For convenience, there
+// is a provider switch to deal with images on DockerHub differently.
+//
+// DockerHub images:
+// - In case the repository only has one tag, the repository is deleted.
+// - If there are multiple tags, the tag to be deleted is overwritten
+//   with an empty image (to remove the content, and save quota).
```

Comment:
What if we have a business tier, should we try to delete the tag?

Reply:
Good question. I do not know.. Tough one. I think even with a business tier, we would have to use a Docker custom API call. Definitely worth the investigation. Do you see that as part of this PR or a separate one?

Reply:
I leave that up-to-you. Problem will be how to test it without a business tier account. :-)